src/main/kotlin/uk/gov/gdx/datashare/controllers/AdminController.kt

```diff
@@ -59,4 +66,39 @@ class AdminController(
     @RequestBody
     cognitoClientRequest: CognitoClientRequest,
   ) = cognitoService.createUserPoolClient(cognitoClientRequest)
+
+  @PostMapping("/acquirer")
+  @Operation(
+    summary = "Create acquirer",
+    description = "Need scope of events/admin",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Cognito client details",
+      ),
+    ],
+  )
+  fun createAcquirer(
+    @Schema(
+      required = true,
+      implementation = CreateAcquirerRequest::class,
+    )
+    @RequestBody
+    createAcquirerRequest: CreateAcquirerRequest,
+  ) = cognitoService.createUserPoolClient(
+    CognitoClientRequest(createAcquirerRequest.clientName, listOf(CognitoClientType.ACQUIRER)),
+  )?.let {
+    val consumer = consumersService.addConsumer(ConsumerRequest(createAcquirerRequest.clientName))
+    consumersService.addConsumerSubscription(
+      consumer.id,
+      ConsumerSubRequest(
+        oauthClientId = it.clientId,
+        eventType = createAcquirerRequest.eventType,
+        enrichmentFields = createAcquirerRequest.enrichmentFields,
+        enrichmentFieldsIncludedInPoll = createAcquirerRequest.enrichmentFieldsIncludedInPoll,
+      ),
+    )
+
+    CognitoClientResponse(it.clientName, it.clientId, it.clientSecret)
```

Comment:
I think  this should be in a service somewhere, and marked with `@Transactional` so we don't end up with half complete data

Reply:
Done